src/Val.cc

```diff
@@ -2398,7 +2398,7 @@ ValPtr TableVal::Remove(const Val& index, bool broker_forward, bool* iterators_i
     if ( broker_forward && ! broker_store.empty() )
         SendToStore(&index, nullptr, ELEMENT_REMOVED);
 
-    if ( change_func ) {
+    if ( change_func && k ) {
         // this is totally cheating around the fact that we need a Intrusive pointer.
         ValPtr changefunc_val = RecreateIndex(*(k.get()));
```

Comment:
Drive-by: If `Val::Clone()` was const we could probably just do `index.Clone()` here.